File: shared/src/test/java/org/pac4j/lagom/javadsl/TestServiceImpl.java
package org.pac4j.lagom.javadsl;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import org.pac4j.core.config.Config;

import javax.inject.Inject;

import static org.pac4j.core.authorization.authorizer.IsAuthenticatedAuthorizer.isAuthenticated;
import static org.pac4j.core.authorization.authorizer.RequireAnyRoleAuthorizer.requireAnyRole;
import static org.pac4j.lagom.javadsl.ClientNames.COOKIE_CLIENT;
import static org.pac4j.lagom.javadsl.ClientNames.HEADER_CLIENT;
import static org.pac4j.lagom.javadsl.ClientNames.HEADER_JWT_CLIENT;

import static java.util.concurrent.CompletableFuture.completedFuture;

/**
 * Implementation of Lagom service for tests.
 *
 * @author Sergey Morgunov
 * @since 1.0.0
 */
public class TestServiceImpl implements TestService, SecuredService {

    private final Config securityConfig;

    @Inject
    public TestServiceImpl(Config securityConfig) {
        this.securityConfig = securityConfig;
    }

    @Override
    public ServiceCall<NotUsed, String> defaultAuthenticate() {
        return authenticate(profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> defaultAuthorize() {
        return authorize(isAuthenticated(), profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> defaultAuthorizeByRole() {
        return authorize(requireAnyRole("role"), profile ->
            request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> defaultAuthorizeConfig() {
        String authorizerName = "_authenticated_";
        return authorize(authorizerName, profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> cookieAuthenticate() {
        return authenticate(COOKIE_CLIENT, profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> cookieAuthorize() {
        return authorize(COOKIE_CLIENT, isAuthenticated(), profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> cookieAuthorizeConfig() {
        String authorizerName = "_authenticated_";
        return authorize(COOKIE_CLIENT, authorizerName, profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> headerAuthenticate() {
        return authenticate(HEADER_CLIENT, profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> headerAuthorize() {
        return authorize(HEADER_CLIENT, isAuthenticated(), profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> headerAuthorizeConfig() {
        String authorizerName = "_authenticated_";
        return authorize(HEADER_CLIENT, authorizerName, profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> headerJwtAuthenticate() {
        return authenticate(HEADER_JWT_CLIENT, profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public Config getSecurityConfig() {
        return securityConfig;
    }
}

File: shared/src/test/scala/org/pac4j/lagom/scaladsl/TestServiceImpl.scala
package org.pac4j.lagom.scaladsl
import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.ServiceCall
import com.lightbend.lagom.scaladsl.server.ServerServiceCall
import org.pac4j.core.authorization.authorizer.IsAuthenticatedAuthorizer.isAuthenticated
import org.pac4j.core.config.Config
import org.pac4j.core.profile.CommonProfile
import org.pac4j.lagom.scaladsl.ClientNames.{HEADER_CLIENT, HEADER_JWT_CLIENT}

import scala.concurrent.Future

/**
  * Implementation of Lagom service for tests.
  *
  * @author Vladimir Kornyshev
  * @since 1.0.0
  */
class TestServiceImpl(override val securityConfig: Config) extends TestService with SecuredService {

  override def defaultAuthenticate: ServiceCall[NotUsed, String] = {
    authenticate((profile: CommonProfile) => ServerServiceCall { _: NotUsed => Future.successful(profile.getId) })
  }

  override def defaultAuthorize: ServiceCall[NotUsed, String] = {
    authorize(isAuthenticated[CommonProfile](), (profile: CommonProfile) => ServerServiceCall { _: NotUsed => Future.successful(profile.getId) })
  }

  override def defaultAuthorizeConfig: ServiceCall[NotUsed, String] = {
    val authorizerName = "_authenticated_"
    authorize(authorizerName, (profile: CommonProfile) => ServerServiceCall { _: NotUsed => Future.successful(profile.getId) })
  }

  override def headerAuthenticate: ServiceCall[NotUsed, String] = {
    authenticate(HEADER_CLIENT, (profile: CommonProfile) => ServerServiceCall { _: NotUsed => Future.successful(profile.getId) })
  }

  override def headerAuthorize: ServiceCall[NotUsed, String] = {
    authorize(HEADER_CLIENT, isAuthenticated[CommonProfile](), (profile: CommonProfile) => ServerServiceCall { _: NotUsed => Future.successful(profile.getId) })
  }

  override def headerAuthorizeConfig: ServiceCall[NotUsed, String] = {
    val authorizerName = "_authenticated_"
    authorize(HEADER_CLIENT, authorizerName, (profile: CommonProfile) => ServerServiceCall { _: NotUsed => Future.successful(profile.getId) })
  }

  override def headerJwtAuthenticate: ServiceCall[NotUsed, String] = {
    authenticate(HEADER_JWT_CLIENT, (profile: CommonProfile) => ServerServiceCall { _: NotUsed => Future.successful(profile.getId) })
  }

}
